pi_cond.c

```diff
@@ -23,77 +26,80 @@ void pi_cond_free(pi_cond_t *cond)
 
 int pi_cond_init(pi_cond_t *cond, struct pi_mutex *mutex, uint32_t flags)
 {
-	pthread_condattr_t attr;
 	struct timespec ts = { 0, 0 };
 	int ret;
 
-	ret = pthread_condattr_init(&attr);
-	if (ret)
-		goto out;
-
-	/* All RTPI condvars are CLOCK_MONOTONIC */
-	ret = pthread_condattr_setclock(&attr, CLOCK_MONOTONIC);
-	if (ret)
+	if (flags & ~(RTPI_COND_PSHARED)) {
+		ret = -EINVAL;
 		goto out;
+	}
 
-	if (flags && RTPI_COND_PSHARED) {
-		ret = pthread_condattr_setpshared(&attr, PTHREAD_PROCESS_SHARED);
-		if (ret)
-			goto out;
+	if (flags & RTPI_COND_PSHARED) {
+		cond->flags = RTPI_COND_PSHARED;
 	}
 
-	ret = pthread_cond_init(&cond->cond, &attr);
-	if (ret)
+	/* PSHARED has to match on both. */
+	if ((cond->flags & RTPI_COND_PSHARED) ^ (mutex->flags % RTPI_MUTEX_PSHARED)) {
+		ret = -EINVAL;
 		goto out;
+	}
 
 	cond->mutex = mutex;
 
-	/* Force association with the specified mutex */
-	ts.tv_nsec = 1;
-	ret = pi_mutex_lock(mutex);
-	if (ret)
-		goto out;
-	ret = pthread_cond_timedwait(&cond->cond, &cond->mutex->futex, &ts);
-	if (ret == ETIMEDOUT)
-		ret = 0;
-	else if (ret == 0)
-		ret = -EINVAL;
-
- out:
+	ret = 0;
+out:
```

Comment:
Please maintain the space to the left of the label, it avoids diff using the label as the function name in future changes.